frame/staking/src/pallet/mod.rs

```diff
@@ -559,6 +625,114 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Wrapper struct for Era related information. It is not a pure encapsulation as these storage
+	/// items can be accessed directly but nevertheless, its recommended to use `EraInfo` for
+	/// accessing the following: (1) `ErasStakers`, (2) `ClaimedRewards`, (3) `ErasStakersClipped`,
+	/// (4) `ErasStakersPaged`, (5) `ErasTotalStake`.
+	// TODO(Ank4n): add all era related storage items in this struct
+	pub(crate) struct EraInfo<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> EraInfo<T> {
+		/// Temporary function which looks at both (1) passed param `T::StakingLedger` for legacy
+		/// non-paged rewards, and (2) `T::ClaimedRewards` for paged rewards. This function can be
+		/// removed once `$HistoryDepth` eras have passed and none of the older non-paged rewards
+		/// are relevant/claimable.
+		// TODO: Cleanup tracking issue: #13034
+		pub(crate) fn is_rewards_claimed_temp(
+			era: EraIndex,
+			ledger: &StakingLedger<T>,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> bool {
+			ledger.legacy_claimed_rewards.binary_search(&era).is_ok() ||
+				Self::is_rewards_claimed(era, validator, page)
+		}
+
+		/// Check if the rewards for the given era and page index have been claimed.
+		///
+		/// This is only used for paged rewards. Once older non-paged rewards are no longer
+		/// relevant, `is_rewards_claimed_temp` can be removed and this function can be made public.
+		fn is_rewards_claimed(era: EraIndex, validator: &T::AccountId, page: PageIndex) -> bool {
+			ClaimedRewards::<T>::get(era, validator).iter().any(|&p| page == p)
+		}
+
+		/// Get exposure info for a validator at a given era and page.
+		///
+		/// This builds a paged exposure from `ExposureOverview` and `ExposurePage` of the
+		/// validator. For older non-paged exposure, it returns the clipped exposure directly.
+		pub(crate) fn get_validator_exposure(
+			era: EraIndex,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> Exposure<T::AccountId, BalanceOf<T>> {
+			return match <ErasStakersPaged<T>>::get(era, (validator, page)) {
+				// only return clipped exposure if page zero and no paged exposure entry
+				None if page == 0 => <ErasStakersClipped<T>>::get(&era, validator),
```

Comment:
is this also temporary?

Reply:
Yes, as long as we keep ErasStakersClipped.